modules/core/include/opencv2/core.hpp

```diff
@@ -3294,6 +3294,12 @@ class CV_EXPORTS_W Algorithm
     void writeFormat(FileStorage& fs) const;
 };
 
+class CV_EXPORTS_W_SIMPLE DeepNeuralNet
```

Comment:
the name is quite confusing. `cv::dnn::Net` vs `cv::DeepNeuralNet`. We can still have it in the core, but use more consistent name, e.g. `cv::dnn::BaseNet`:

```
namespace dnn {
class CV_EXPORTS_W_SIMPLE NetBase
{
public:
      virtual ~NetBase();
};
}
```